### DISCUSSION TITLE
Customize Window

### DISCUSSION THREAD

**Discussion by AliShahesmaeili** [microsoft/WindowsAppSDK#2027] [Q&A] [answered]
How to customize window style? like Voice typing in Windows 11. Is there an example?

Inactive:
![image](https://user-images.githubusercontent.com/67000626/151213137-c169a0fa-6a4b-4708-8130-68bf9ee3a8b0.png)

Active:
![image](https://user-images.githubusercontent.com/67000626/151213277-7217df33-46a8-45c1-9d65-17667e45f828.png)

**Comment by castorix** [comment, marked as answer]
What do you mean by "customize" ?
With rounded corners ? , like :  [Apply rounded corners in desktop apps for Windows 11](https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/apply-rounded-corners)

**Reply by AliShahesmaeili** [reply]
In wpf I can customize my window style (buttons style, size, rounded corners, color, etc.)
But in UWP there was a limit to doing this. Now in WinAppSDK, like wpf, can windows be completely customized?

**Reply by castorix** [reply]
Windows App SDK applications are Desktop apps, so Win32 regions can be used ([SetWindowRgn](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowrgn) and so on)

A test with a Region from a Butterfly image : 

![SetWindowRgn_WinUI](https://user-images.githubusercontent.com/22345506/151671096-f8821d5e-49a1-4ad9-8557-9f8e7d1e654d.jpg)
